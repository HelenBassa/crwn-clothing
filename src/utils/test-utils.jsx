import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };