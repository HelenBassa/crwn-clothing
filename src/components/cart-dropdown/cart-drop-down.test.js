import { render, screen } from "../../utils/test-utils";
import CartDropdown from "./cart-drop-down.component";

test("should render CartDropdown component", () => {
  const { container } = render(<CartDropdown />);
  expect(container).toMatchSnapshot();
});

test("should render EmptyMessageContainer if cartItems is empty", () => {
  render(<CartDropdown />);
  expect(screen.queryByRole("list")).toBeNull();
  expect(screen.getByText("Your cart is empty")).toBeInTheDocument();
});
