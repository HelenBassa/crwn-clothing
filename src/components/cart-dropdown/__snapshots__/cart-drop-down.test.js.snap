// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartDropdown component 1`] = `
<div>
  <div
    class="sc-ftTHYK cZLpSt"
  >
    <div
      class="sc-kDvujY dUa-DAJ"
    >
      <span
        class="sc-jrcTuL huYouY"
      >
        Your cart is empty
      </span>
    </div>
    <button
      class="sc-dkrFOg KWCcW"
    >
      GO TO CHECKOUT
    </button>
  </div>
</div>
`;
